src/popup/App.tsx

```diff
@@ -20,127 +20,145 @@ import emptyInbox from './assets/empty_inbox.svg';
+            return <Text>Fetching content...</Text>;
+        }
 
-    if (mrList === fetchingContentState) {
-        sendMsg(null);
-    }
+        if (appStatus === 'error') {
+            return (
+                <Flash m={2} scheme="red">
+                    {errorMessage}
+                </Flash>
+            );
+        }
+
+        // show data
+        let mrList = [];
+        if (currentTab === 0) {
+            mrList = mrData.mrAssigned;
+        } else {
+            mrList = mrData.mrGiven;
+        }
+
+        if (!mrList || mrList.length === 0) {
+            return <img src={emptyInbox} className={'emptyInbox'} />;
+        }
+        return (
+            <FilterList className={'mrList'}>
+                {mrList.map((mr: MergeRequestsDetails) => (
+                    <MergeRequest mr={mr} key={mr.id} />
+                ))}
+            </FilterList>
+        );
+    }, [appStatus, mrData, currentTab, errorMessage]);
+
+    const getMrRatio = useCallback(() => {
+        if (appStatus !== 'success' || mrData === null) {
+            return 100;
+        }
+        let mrList = [];
+        if (currentTab === 0) {
+            mrList = mrData.mrAssigned;
+        } else {
+            mrList = mrData.mrGiven;
+        }
+        let rate = 0;
+        if (mrList && mrList.length > 0) {
+            rate = (mrList.length - mrData.mrToReview) / mrList.length;
+        }
+        return Math.floor(rate * 100);
+    }, [appStatus, mrData, currentTab]);
 
     return (
         <ThemeProvider theme={primer}>
             <div className={'container'}>
                 <TabNav aria-label="Main" mb={2}>
-                    <TabNav.Link
-                        onClick={sendMsg}
-                        data-key={'getMRs'}
-                        data-id={'0'}
-                        href="#MR"
-                        className={tabSelected === '0' ? 'selected' : ''}
-                    >
+                    <TabNav.Link onClick={() => setCurrentTab(0)} className={currentTab === 0 ? 'selected' : ''}>
                         To Review{' '}
                         <Label variant="small" bg="#dc3545">
-                            {mrToReview}
+                            {mrData ? mrData.mrToReview : 0}
                         </Label>
                     </TabNav.Link>
-                    <TabNav.Link
-                        onClick={sendMsg}
-                        data-key={'getMyMRs'}
-                        data-id={'1'}
-                        href="#MyMR"
-                        className={tabSelected === '1' ? 'selected' : ''}
-                    >
+                    <TabNav.Link onClick={() => setCurrentTab(1)} className={currentTab === 1 ? 'selected' : ''}>
                         Under Review{' '}
-                        <Tooltip aria-label={mrReviewed + ' have been reviewed'} direction="s">
+                        <Tooltip aria-label={mrData.mrReviewed + ' have been reviewed'} direction="s">
```

Comment:
Not sure it is intended change but we had "x / y have been reviewed" before.

Reply:
not intended!